packages/firebase_core/firebase_core/lib/src/firebase_app.dart

```diff
@@ -14,37 +14,23 @@ class FirebaseApp {
   static final String defaultAppName =
       Platform.isIOS ? '__FIRAPP_DEFAULT' : '[DEFAULT]';
 
-  @visibleForTesting
-  static const MethodChannel channel = MethodChannel(
-    'plugins.flutter.io/firebase_core',
-  );
-
   /// A copy of the options for this app. These are non-modifiable.
   ///
   /// This getter is asynchronous because apps can also be configured by native
   /// code.
   Future<FirebaseOptions> get options async {
-    final Map<String, dynamic> app =
-        await channel.invokeMapMethod<String, dynamic>(
-      'FirebaseApp#appNamed',
-      name,
-    );
+    final PlatformFirebaseApp app =
+        await FirebaseCorePlatform.instance.appNamed(name);
     assert(app != null);
-    return FirebaseOptions.from(app['options']);
+    return app.options;
   }
 
   /// Returns a previously created FirebaseApp instance with the given name,
   /// or null if no such app exists.
   static Future<FirebaseApp> appNamed(String name) async {
-    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-    // https://github.com/flutter/flutter/issues/26431
-    // ignore: strong_mode_implicit_dynamic_method
-    final Map<String, dynamic> app =
-        await channel.invokeMapMethod<String, dynamic>(
-      'FirebaseApp#appNamed',
-      name,
-    );
-    return app == null ? null : FirebaseApp(name: app['name']);
+    final PlatformFirebaseApp app =
+        await FirebaseCorePlatform.instance.appNamed(name);
+    return app == null ? null : FirebaseApp(name: app.name);
```

Comment:
I think you can express this as:

```dart
return app ?? FirebaseApp(name: app.name);
```

Reply:
Also, why doesn't the `appNamed` method ensure that it always returns an app, instead of having this code in the plugin? What does it mean that the instance returns a `null` app?

Reply:
I read the above TOTALLY backwards 🤦‍♂

Reply:
That won't work because then it would return a `PlatformFirebaseApp`. I think it returns `null` when there is no app with the given `name`.

Reply:
Makes way more sense than what I was "understanding" :P